packages/contracts/hardhat/tasks/check-balance-for-deploy.ts

```diff
@@ -0,0 +1,122 @@
+import fs from 'node:fs'
+import path from 'node:path'
+import { TASK_DEPLOY } from 'hardhat-deploy'
+import { task } from 'hardhat/config'
+import type { HardhatRuntimeEnvironment } from 'hardhat/types'
+import type { DeployFunction } from 'hardhat-deploy/types'
+
+export const checkBalanceForDeploy = task(TASK_DEPLOY, async (_args, hre: HardhatRuntimeEnvironment, runSuper) => {
+  if (hre.network.name === 'hardhat') {
+    return runSuper(_args)
+  }
+  const gasUsedTotal = await getEstimatedDeploymentGas(hre)
+  const gasPrice = await hre.ethers.provider.getGasPrice()
+  const estimatedDeploymentPrice = gasPrice.mul(gasUsedTotal).mul(120).div(100)
```

Comment:
why 120 and 100?

Reply:
Can you add comment or use some named const to be clear for that. Looks like magic numbers which scary to change right now